drivers/modem/gsm_ppp.c

```diff
@@ -84,6 +84,9 @@ static struct gsm_modem {
 	bool mux_setup_done : 1;
 	bool setup_done : 1;
 	bool attached : 1;
+
+	void (*modem_on_cb)(void *user_data);
```

Comment:
Overall looks good. I would put device pointer (same as what is given when registering the callback) as a parameter here. It is currently not used for anything, but I have seen in the past and with similar usage patterns that usually the user_data is not enough and some more context is needed.
So I would do here
```
void (*modem_on_cb)(const struct device *dev, void *user_data);
void (*modem_off_cb)(const struct device *dev, void *user_data);
```

Reply:
Done.